src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -0,0 +1,284 @@
+                # basic auth
+                # if endpoint.is_secure:
+                #     security = '{ "basicAuth": [] }'
+                # else:
+                #     security = ''
+                security = ''
+
+                methods += """
+                "{}": {{
+                    "tags": ["{}"],
+                    "summary": "",
+                    "consumes": [
+                        "application/json"
+                    ],
+                    "produces": [
+                        "application/json"
+                    ],
+                    "parameters": [{}],
+                    {},
+                    "security": [{}]
+                }},
+                """.format(method.lower(), endpoint.group, ",".join(params),
+                           self._gen_responses_descriptions(method), security)
+            if not skip:
+                paths += """
+                "{}": {{ {} }},
+                """.format(path[len(baseUrl):], methods[:-1])
+
+        if not baseUrl:
+            baseUrl = "/"
+        spec = """
+        {{
+            "swagger": "2.0",
+            "info": {{
+                "description": "Please note that this API is not an official \
+Ceph REST API to be used by third-party applications. It's primary purpose is \
+to serve the requirements of the Ceph Dashboard and is subject to change at \
+any time. Use at your own risk.",
+                "version": "v1",
+                "title": "Ceph-Dashboard REST API"
+            }},
+            "host": "{}",
+            "basePath": "{}",
+            "tags": [{}],
+            "schemes": ["https"],
+            "paths": {{
+                {}
+            }},
+        }}
+        """.format(host, baseUrl, self._gen_tags(all_endpoints), paths[:-1])
+
+        return spec
+
+    @Endpoint(json_response=False)
+    def __call__(self, all_endpoints=False):
+        page = """
+        <!DOCTYPE html>
+        <html>
+        <head>
+            <meta charset="UTF-8">
+            <link href="https://fonts.googleapis.com/css?family=Open+Sans:400, \
+                        700|Source+Code+Pro:300,600|Titillium+Web:400,600,700"
+                  rel="stylesheet">
+            <link rel="stylesheet" type="text/css"
+                  href="//unpkg.com/swagger-ui-dist@3/swagger-ui.css" >
+            <style>
+                html
+                {{
+                    box-sizing: border-box;
+                    overflow: -moz-scrollbars-vertical;
+                    overflow-y: scroll;
+                }}
+                *,
+                *:before,
+                *:after
+                {{
+                    box-sizing: inherit;
+                }}
+
+                body {{
+                    margin:0;
+                    background: #fafafa;
+                }}
+            </style>
+        </head>
+        <body>
+
+        <div id="swagger-ui"></div>
+
+        <script src="//unpkg.com/swagger-ui-dist@3/swagger-ui-bundle.js">
```

Comment:
Same here, the JS file must be shipped and installed by Ceph Dashboard.

Reply:
I wonder if we can add `noreferrer` here, too?

Reply:
the `script` tag does not support the `rel` attribute